src/radical/pilot/agent/scheduler/base.py

```diff
@@ -653,18 +653,29 @@ def _schedule_waitpool(self):
         # We define `tuple_size` as
         #     `(cpu_processes + gpu_processes) * cpu_threads`
         #
-        tasks = list(self._waitpool.values())
-        tasks.sort(key=lambda x:
+        tasks   = list(self._waitpool.values())
+        to_wait = list()
+        to_test = list()
+
+        for task in tasks:
+            named_env = task['description'].get('named_env')
+            if named_env:
+                if os.path.exists('%s.ok' % named_env):
+                    to_test.append(task)
+                else:
+                    to_wait.append(task)
+
+        to_test.sort(key=lambda x:
                 (x['tuple_size'][0] + x['tuple_size'][2]) * x['tuple_size'][1],
                  reverse=True)
 
         # cycle through waitpool, and see if we get anything placed now.
-        scheduled, unscheduled = ru.lazy_bisect(tasks,
+        scheduled, unscheduled = ru.lazy_bisect(to_test,
```

Comment:
I got a little bit confused, if there is no tasks with `named_env` set, then all that tasks will not reach here?